src/main/java/seedu/linkedout/model/ModelManager.java

```diff
@@ -144,7 +150,52 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return linkedout.equals(other.linkedout)
                 && userPrefs.equals(other.userPrefs)
-                && filteredApplicants.equals(other.filteredApplicants);
+                && filteredApplicants.equals(other.filteredApplicants)
+                && sortedApplicants.equals(other.sortedApplicants);
+    }
+
+    //=========== Sorted Applicant List Accessors =============================================================
+
+    @Override
+    public ObservableList<Applicant> getSortedApplicantList() {
+        return sortedApplicants;
+    }
+
+    //=========== Search Applicant List Accessors =============================================================
+
+    @Override
+    public void updateSearchedApplicantList(List<KeywordsPredicate> predicates) {
+        requireNonNull(predicates);
+        Predicate<Applicant> keywordPredicate = combinePredicates(predicates);
+        updateFilteredApplicantList(keywordPredicate);
+        sortedApplicants.setComparator((a1, a2) -> numberOfKeywordMatches(a2, predicates)
+                - numberOfKeywordMatches(a1, predicates));
     }
 
+    /**
+     * Returns the number of matched input keywords with an applicant
+     * @param applicant
+     * @param predicates list of predicates
+     * @return number of keywords matched
+     */
+    private static int numberOfKeywordMatches (Applicant applicant, List<KeywordsPredicate> predicates) {
+        int matchedNumber = 0;
+        for (int i = 0; i < predicates.size(); i++) {
+            matchedNumber += predicates.get(i).numberOfKeywordMatches(applicant);
+        }
+        return matchedNumber;
+    }
+
+    /**
+     * Return the combined predicates of a short-circuiting logical AND of given predicates
+     * @param predicateList list of predicates
+     * @return combined predicate
+     */
+    private static Predicate<Applicant> combinePredicates(List<KeywordsPredicate> predicateList) {
+        Predicate<Applicant> predicates = predicateList.get(0);
```

Comment:
Same here is there a need to check

Reply:
same as above but ill add checking for both methods incase these methods are being called somewhere else